### ISSUE TITLE
Astropy.test() fails on OS 10.11.3 and python3.5

### ISSUE THREAD

**Issue by Yuehchia** [astropy/astropy#5639]
```
______________________________________________________________________ test_skyoffset[inradec1-expectedlatlon1-tolsep1] ______________________________________________________________________

inradec = <Quantity [ 45.,  0.] deg>, expectedlatlon = <Quantity [  0.,-45.] deg>, tolsep = <Quantity 0.001 arcsec>, originradec = <Quantity [ 45., 45.] deg>

    @pytest.mark.parametrize("inradec,expectedlatlon, tolsep", [
        ((45, 45)*u.deg, (0, 0)*u.deg, .001*u.arcsec),
        ((45, 0)*u.deg, (0, -45)*u.deg, .001*u.arcsec),
        ((45, 90)*u.deg, (0, 45)*u.deg, .001*u.arcsec),
        ((46, 45)*u.deg, (1*np.cos(45*u.deg), 0)*u.deg, 16*u.arcsec),
        ])
    def test_skyoffset(inradec, expectedlatlon, tolsep, originradec=(45, 45)*u.deg):
        origin = ICRS(*originradec)
>       skyoffset_frame = SkyOffsetFrame(origin=origin)

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/coordinates/tests/test_skyoffset_transformations.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/coordinates/builtin_frames/skyoffset.py:199: in __new__
    newcls = make_skyoffset_cls(origin_frame.__class__)
/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/coordinates/builtin_frames/skyoffset.py:110: in make_skyoffset_cls
    {'__doc__': SkyOffsetFrame.__doc__})
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'astropy.coordinates.builtin_frames.skyoffset.make_skyoffset_cls.<locals>.SkyOffsetMeta'>, name = 'SkyOffsetFrame'
bases = (<class 'astropy.coordinates.builtin_frames.skyoffset.SkyOffsetFrame'>, <class 'astropy.coordinates.builtin_frames.icrs.ICRS'>)
members = {'__doc__': '\n    A frame which is relative to some specific position and oriented to match\n    its frame.\n\n    Sk...='lat', framename='dec', defaultunit='recommended')]}, 'default_representation': <property object at 0x1046d11d8>, ...}

    def __new__(cls, name, bases, members):
        # Only 'origin' is needed here, to set the origin frame properly.
        members['origin'] = CoordinateAttribute(frame=framecls, default=None)
    
        # This has to be done because FrameMeta will set these attributes
        # to the defaults from BaseCoordinateFrame when it creates the base
        # SkyOffsetFrame class initially.
        members['_frame_specific_representation_info'] = framecls._frame_specific_representation_info
        members['_default_representation'] = framecls._default_representation
    
        newname = name[:-5] if name.endswith('Frame') else name
        newname += framecls.__name__
    
        res = super(SkyOffsetMeta, cls).__new__(cls, newname, bases, members)
    
        # now go through all the component names and make any spherical names be "lon" and "lat"
        # instead of e.g. "ra" and "dec"
    
        lists_done = []
        for nm, component_list in res._frame_specific_representation_info.items():
            if nm in ('spherical', 'unitspherical'):
                gotlatlon = []
                for i, comp in enumerate(component_list):
                    if component_list in lists_done:
                        # we need this because sometimes the component_
                        # list's are the exact *same* object for both
                        # spherical and unitspherical.  So looping then makes
                        # the change *twice*.  This hack bypasses that.
                        continue
    
                    if comp.reprname in ('lon', 'lat'):
                        dct = namedtuple_asdict(comp)
                        # this forces the component names to be 'lat' and
                        # 'lon' regardless of what the actual base frame
                        # might use
                        dct['framename'] = comp.reprname
>                       component_list[i] = type(comp)(**dct)
E                       TypeError: __new__() missing 1 required positional argument: 'reprname'

/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy/coordinates/builtin_frames/skyoffset.py:95: TypeError


Total '24 failed, 8150 passed, 127 skipped, 37 xfailed in 280.39 seconds'.
Those failed tests all notified 'TypeError: __new__() missing 1 required positional argument: 'reprname'.
```

I ran the test on:

```
platform darwin -- Python 3.5.0, pytest-2.8.3, py-1.4.30, pluggy-0.3.1

Running tests with Astropy version 1.2.1.
Running tests in /Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/astropy.

Date: 2016-12-22T21:23:28

Platform: Darwin-15.3.0-x86_64-i386-64bit

Executable: /Library/Frameworks/Python.framework/Versions/3.5/bin/python3

Full Python Version: 
3.5.0 (v3.5.0:374f501f4567, Sep 12 2015, 11:00:19) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]

encodings: sys: utf-8, locale: US-ASCII, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.11.3
Scipy: 0.18.1
Matplotlib: 1.5.3
h5py: 2.6.0
Pandas: 0.19.1
```

**Comment by pllim** [comment]
Just for the record, do you see the same failures in the newly released 1.3?

**Comment by tcs3475** [comment]
I have encountered a similar problem running astropy 1.3 and Python 3.5 on OS X 10.10. This is similar to the error described in #5079. However, they are running python 3.4.x. I managed to get my version of astropy to pass the 24 tests it had previously failed by editing the alteration that was used to fix #5079. In the misc.py file located within /astropy/utils/compat, I changed line 159 from,
if sys.version_info[0] == 3 and sys.version_info[:3] < (3, 4, 4)
to,
if sys.version_info[0] == 3 and sys.version_info[:3] < (3, 5, 4)
I'm not sure if this was a good way to correct the issue though.
